kivy/uix/widget.py

```diff
@@ -1053,6 +1053,13 @@ def get_window_matrix(self, x=0, y=0):
         Keep in mind that the `width` property is subject to layout logic and
         that this has not yet happened at the time of the widget's `__init__`
         method.
+
+    .. warning::
+        Kivy allows the `width` property to be negative, but Kivy does not
+        support negative sizes. It is recommended to always have a positive
+        `width` property. If not, the programmer must account for the
+        consequences of a negative `width`. For example, `right` will actually
+        be left of `x` when `width` is negative.
```

Comment:
```suggestion

    .. warning::
        A negative size is not supported.
```

I think something very concise is better because we don't need say anything beyond it not being supported. And most people will not care much beyond that.